libs/crypto/keys_manager_impl.cpp

```diff
@@ -20,39 +20,85 @@
+   * specified field in given keypair
    * @tparam T - keypair field type
    * @tparam V - value type to deserialize
    * @param field - keypair field to be deserialized
    * @param value - value to be deserialized
-   * @return keypair on success, otherwise nullopt
+   * @return function that will return keypair on success, otherwise nullopt
    */
   template <typename T, typename V>
-  auto deserializeKeypairField(T iroha::keypair_t::*field, const V &value) {
+  auto deserializeKeypairField(T keypair_t::*field, const V &value) {
     return [=](auto keypair) mutable {
-      return iroha::hexstringToArray<T::size()>(value)
-          | iroha::assignObjectField(keypair, field);
+      return hexstringToArray<T::size()>(value)
+          | assignObjectField(keypair, field);
+    };
+  }
+
+  /**
+   * Function for the private key encryption via XOR
+   * @param privkey is a private key
+   * @param pass_phrase is a key for encryption
+   * @return encrypted string
+   */
+  std::string encrypt(const privkey_t &privkey,
+                      const std::string &pass_phrase) {
+    std::string ciphertext(privkey.size(), '\0');
+    const auto pass_size = std::max(1ul, pass_phrase.size());
+
+    for (auto i = 0u; i < privkey.size(); i++) {
+      ciphertext[i] = privkey[i] ^ pass_phrase[i % pass_size];
+    }
+    return ciphertext;
+  }
+
+  /**
+   * Return a function which will try to deserialize and then decrypt private
+   * key via XORing with pass phrase
+   * @param s is an encrypted data from file
+   * @param pass_phrase for decryption
+   * @return function that will set keypair::privkey on successful
+   *         deserialization and decryption
+   */
+  auto deserializedEncrypted(const std::string &s,
+                             const std::string &pass_phrase) {
+    constexpr auto size = privkey_t::size();
+    auto decryption = [=](auto binstr) {
+      std::string key(binstr.size(), '\0');
+      const auto pass_size = std::max(1ul, pass_phrase.size());
+
+      for (auto i = 0u; i < binstr.size(); i++) {
+        key[i] = binstr[i] ^ pass_phrase[i % pass_size];
+      }
+      return nonstd::make_optional(key);
+    };
+
+    return [=](auto keypair) mutable {
+      return hexstringToBytestring(s) | decryption
+          | stringToBlob<
+                 size> | assignObjectField(keypair, &keypair_t::privkey);
     };
   }
 
   KeysManagerImpl::KeysManagerImpl(const std::string &account_name)
       : account_name_(std::move(account_name)),
         log_(logger::log("KeysManagerImpl")) {}
 
-  bool KeysManagerImpl::validate(const iroha::keypair_t &keypair) const {
+  bool KeysManagerImpl::validate(const keypair_t &keypair) const {
     std::string test = "12345";
-    auto sig = iroha::sign(
-        iroha::sha3_256(test).to_string(), keypair.pubkey, keypair.privkey);
-    if (not iroha::verify(
-            iroha::sha3_256(test).to_string(), keypair.pubkey, sig)) {
+    auto sig =
```

Comment:
Looks like randomString() from generator.cpp can be used.

Reply:
Quite out of that pr, anyway that's easy to fix